### PR TITLE
Page inside a page

### PR DESCRIPTION
**Pull request by maheshneogi**
- Made changes to be able to add page inside a page
- Made changes to render page block inside page which shows the title
- Made changes to navigate to the page when clicked on the page block

### CODE REVIEW

File: app/src/types/BlockFactory.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export const createBlock = (blockType: BlockType): Block => {
 const createDefaultProperties = (blockType: BlockType): BlockProperties => {
   switch (blockType) {
     case BlockType.Page:
-      return {title: ''}
+      return {title: 'untitled-page'}
```
Comment (anishbhanwala) [inline app/src/types/BlockFactory.ts:19]: The default value should be empty. If we want to show 'Untitled' if the title is empty then that should be handled while displaying it.

File: app/src/components/content/ContentPointer.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+export const ContentPointer: React.FC<Props> = ({
+    onElementClick,
+    displayText,
+    ...props
+}) => {
+    //TODO: remove inline css
+    return (
+    <div style={{cursor: 'pointer'}} className='content-pointer' onClick={onElementClick}>{displayText}</div>)
```
Comment (anishbhanwala) [inline app/src/components/content/ContentPointer.tsx:15]: I think this should be a link with href to the page. E.g. 
```
<a href="#/pages/page-uid">{displayText || 'Untitled'}</a>
```

Accordingly, the component can be called `PageLink` or `LinkComponent`.

File: app/src/components/content/ContentPointer.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import React from 'react'
+
+interface Props {
+    onElementClick(): void
```
Comment (anishbhanwala) [inline app/src/components/content/ContentPointer.tsx:4]: This can be simply `onClick`

File: app/src/components/content/ContentPointer.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    displayText: string | undefined
```
Comment (anishbhanwala) [inline app/src/components/content/ContentPointer.tsx:5]: You can also do:
`displayText?: string`

**Review by anishbhanwala — CHANGES_REQUESTED** [review]

**Comment by anishbhanwala** [comment]
Also, github checks are failing.

**Comment by maheshneogi** [comment]
@anishbhanwala done with the suggested changes

**Review by anishbhanwala — APPROVED** [review]

**Pull request merged**